File: 739-DailyTemperatures/739-DailyTemperatures.cpp
// Last updated: 6/10/2026, 11:48:02 AM
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> ans(n,0);
        stack<pair<int,int>> st;
        for(int i = n-1 ; i >= 0; i--){
            if(st.empty()) {
                st.push({temperatures[i],i});
                ans[i] = 0;
                continue;
            }
            else{
                while(st.top().first<=temperatures[i]) {
                    st.pop();
                    // cout<<"hi"<<i<<"\n";
                    if(st.empty()) break;
                }
                if(st.empty()) {
                    st.push({temperatures[i],i});
                    ans[i] = 0;
                    continue;
                }
                else{
                    ans[i] = abs(i - st.top().second);
                    st.push({temperatures[i],i});
                }
            }
        }
        return ans;
    }
};

File: 739-DailyTemperatures/739-DailyTemperatures.java
// Last updated: 6/10/2026, 11:48:07 AM
import java.util.Stack;
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack <Integer> stack = new Stack<Integer>();
        int []answer = new int[temperatures.length];
        for(int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int index = stack.pop();
                answer[index] = i - index;
            }
            stack.push(i);
        }
        return answer;         
    }
}